### ISSUE TITLE
Is this logic judgement reversed?

### ISSUE THREAD

**Issue by ZhangYuef** [necla-ml/Diff-JPEG#2]
In `Diff-JPEG-main/diff_jpeg/clipping.py` line 56:
https://github.com/necla-ml/Diff-JPEG/blob/ca31533412eba853760cb695b679e81cfdb67eb9/diff_jpeg/clipping.py#L56

should this be:
```python
grad: Tensor = torch.where(torch.logical_or(input <= ctx.min, input >= ctx.max), 1.0, ctx.scale) * grad_output
```

**Comment by ChristophReich1996** [comment]
Thanks for pointing this out! Fixed with commit a3596bb.

**Comment by ZhangYuef** [comment]
> Thanks for pointing this out! Fixed with commit [a3596bb](https://github.com/necla-ml/Diff-JPEG/commit/a3596bb0117dd8f03a5129489c46d08c18223609).

@ChristophReich1996 Thanks for this quick reply. But I am wondering if this would effect the results reported in the manuscript.

**Comment by ChristophReich1996** [comment]
@ZhangYuef valid concern! If this has an impact this would only affect the backward performance results of our STE approach. I believe that his bug does not affect the sign of the gradient and as we use FGSM and IFGSM results should hopefully not be affected. In the case of Tab. 4, where we analyze the gradient magnitude this bug might have an impact. I will re-run the respective experiment to validate this! In case there are any changes to the results we will fix this in the arXiv version. Again thanks for pointing out this issue!

**Comment by ZhangYuef** [comment]
> @ZhangYuef valid concern! If this has an impact this would only affect the backward performance results of our STE approach. I believe that his bug does not affect the sign of the gradient and as we use FGSM and IFGSM results should hopefully not be affected. In the case of Tab. 4, where we analyze the gradient magnitude this bug might have an impact. I will re-run the respective experiment to validate this! In case there are any changes to the results we will fix this in the arXiv version. Again thanks for pointing out this issue!

Thanks for answering! I agree with your analysis and am curious to see the experiment results. 

Good job!

**Comment by ChristophReich1996** [comment]
@ZhangYuef So I have some updates. As expected this bug had only a very minor effect on the FGSM and IFGSM results. The gradient norms in Tab. 4 have changed.
<img width="583" alt="Screenshot 2023-11-16 at 19 09 17" src="https://github.com/necla-ml/Diff-JPEG/assets/34400551/998b15a1-8d3b-4c98-a2d3-f4bffa94cd8b">
We have updated the arXiv version. The new version should be online in the upcoming days. If you want to check out the results immediately please have a look at my thesis [here](https://christophreich1996.github.io/pdfs/Master_Thesis_Christoph_Reich.pdf).
Thanks again for your help! Cheers Christoph

**Comment by ZhangYuef** [comment]
**Again,** thanks and cheers for your great work : )
